File: 2022/tests/day08.test.ts
import { readFileSync } from 'fs';
import { part1, part2 } from '../src/day08';

const dayNumber = __filename.match(/day(\d+)\.test\.ts/)?.[1];
const inputFile: string = `input/day${dayNumber}.txt`;

const example: string = `
30373
25512
65332
33549
35390
`.trim();
const input: string = readFileSync(inputFile, 'utf8');

describe('part 1', () => {
  test('example', () => {
    expect(part1(example)).toBe(21);
  });
  test(`solution`, () => {
    expect(part1(input)).toBe(1851);
  });
});

describe('part 2', () => {
  test('example', () => {
    expect(part2(example)).toBe(8);
  });
  test(`solution`, () => {
    expect(part2(input)).toBe(574080);
  });
});

File: 2024/AdventOfCode/Tests/Day08Test.cs
using NUnit.Framework;

namespace AdventOfCode.Tests;

[TestFixture]
public class Day08Test {
    private const string Example = "Inputs/08-Example.txt";
    private const string Example2 = "Inputs/08-Example2.txt";
    private const string Input = "Inputs/08.txt";
    private static Day08 _day;

    [TestFixture]
    public class Part1 {
        [Test]
        public void TestExample() {
            _day = new Day08(Example);
            const string expected = "14";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day08(Input);
            const string expected = "344";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }

    [TestFixture]
    public class Part2 {
        [Test]
        public void TestExample() {
            _day = new Day08(Example);
            const string expected = "34";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestExample2() {
            _day = new Day08(Example2);
            const string expected = "9";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day08(Input);
            const string expected = "1182";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }
}